### PR TITLE
add dts support for pistachio_marduk

### LINKED ISSUE #11: Add device tree
For pistachio

### PR DESCRIPTION
**Pull request by abhimanyuv1**
This currently enable initial dts checkin. Also add nand boot and audio support.
This fixes #11

### CODE REVIEW

**Comment by Avinash95** [comment]
verified

**Comment by ghost** [comment]
The only issue now is that the dts is missing the wifi alias.

**Review by abhijit-mahajani-imgtec — CHANGES_REQUESTED** [review]
I think pistachio.dtsi is also missing the change
https://github.com/CreatorDev/openwrt/blob/ci40/target/linux/pistachio/patches-4.1/0062-pistachio.dtsi-add-usb_pwr_on_pins-in-pinctrl.patch

This was required to fix adapter support for Ci40. i.e. booting from USB/Adapter please refer internal ticket https://issues.ba.imgtec.org/browse/KIN-805

**Review by abhijit-mahajani-imgtec — CHANGES_REQUESTED** [review]
pistachio_marduk.dts is missing the change for usb_phy support
https://github.com/CreatorDev/openwrt/blob/ci40/target/linux/pistachio/patches-4.1/0063-marduk_dts-enable-vbus-drive-for-usb_phy.patch

File: arch/mips/boot/dts/img/pistachio_marduk.dts [hunk 1]
```diff
@@ -90,6 +90,17 @@
 		reg = <0>;
 		spi-max-frequency = <50000000>;
 	};
```
Comment (abhijit-mahajani-imgtec) [inline arch/mips/boot/dts/img/pistachio_marduk.dts:125]: Its missing this commit https://github.com/CreatorDev/openwrt/blob/ci40/target/linux/pistachio/patches-4.1/0097-dts-set-linux-mtd-name-for-flash-devices.patch

**Comment by abhijit-mahajani-imgtec** [comment]
> The only issue now is that the dts is missing the wifi alias.

Since this PR is just about basic platform support (not including WiFi), i think it should be ok. WiFi alias and all uccp related changes can be added in subsequent PR.

**Comment by abhimanyuv1** [comment]
Wifi is not included as i didnt had way to test with buildroot without adding uccp module package in it. I will do it in next creator time (i am fine if anyone else want to do it) 

>pistachio_marduk.dts is missing the change for usb_phy support

usb_phy changes are included in uboot so i expect that once ubus power is on in uboot, it will be on in kernel too. I would need to verify it though. if someone can verify it easily then let me know if it is failing.

**Review by abhijit-mahajani-imgtec — APPROVED** [review]

**Pull request merged**